default.nix

```diff
@@ -0,0 +1,116 @@
+{ stdenvNoCC, writeShellScriptBin, writeText, runCommand,
+  stdenv, fetchurl, makeWrapper, nodejs-10_x, yarn2nix }:
+with stdenv.lib; let
+  inherit (builtins) fromJSON toJSON split removeAttrs;
+
+  depsToFetches = deps: concatMap depToFetch (attrValues deps);
+
+  depFetchOwn = { resolved, integrity, ... }:
+    let
+      ssri      = split "-" integrity; # standard subresource integrity
+      hashType  = head ssri;
+      hash      = elemAt ssri 2;
+      fname     = baseNameOf resolved;
+    in nameValuePair resolved (fetchurl {
+      url           = resolved;
+      "${hashType}" = hash;
+      name          = if hasSuffix ".tgz" fname || hasSuffix ".tar.gz" fname
+                      then fname else fname + ".tgz";
+    });
+
+  depToFetch = args @ { resolved ? null, dependencies ? {}, ... }:
+    (optional (resolved != null) (depFetchOwn args)) ++ (depsToFetches dependencies);
+
+  npmCacheInput = lock: writeText "npm-cache-input.json" (toJSON (listToAttrs (depToFetch lock)));
+
+  patchShebangs = writeShellScriptBin "patchShebangs.sh" ''
+    set -e
+    source ${stdenvNoCC}/setup
+    patchShebangs "$@"
+  '';
+
+  shellWrap = writeShellScriptBin "npm-shell-wrap.sh" ''
+    set -e
+    if [ ! -e .shebangs_patched ]; then
+      ${patchShebangs}/bin/patchShebangs.sh .
+      touch .shebangs_patched
+    fi
+    exec bash "$@"
+  '';
+in
+  args @ { src, useYarnLock ? false, yarnIntegreties ? {},
+           npmBuild ? "npm ci", npmBuildMore ? "",
+           buildInputs ? [], npmFlags ? [], ... }:
+    let
+      pkgJson         = src + "/package.json";
+      pkgLockJson     = src + "/package-lock.json";
+      yarnLock        = src + "/yarn.lock";
+
+      yarnIntFile     = writeText "integreties.json" (toJSON yarnIntegreties);
+      lockfile        = if useYarnLock then mkLockFileFromYarn else pkgLockJson;
+
+      info            = fromJSON (readFile pkgJson);
+      lock            = fromJSON (readFile lockfile);
+
+      name            = "${info.name}-${info.version}";
+      npm             = "${nodejs-10_x}/bin/npm";
+      npmAlias        = ''npm() { ${npm} "$@" $npmFlags; }'';
```

Comment:
it's possible to pass npm arguments using env variables, but I didn't want to break nix-shell. Given sufficient divergence between shell and build, maybe they should be split.

Reply:
Could you elaborate on that?